### DISCUSSION TITLE
Are high-bandwidth isochronous endpoints supported?  If not, how hard would it be?

### DISCUSSION THREAD

**Discussion by ccrome** [hathach/tinyusb#1468] [General]
Hi there,
   We're working on a board can can take in a massive number of microphones, and would like to be able stream them at a high sample rate too.  The standard 1024 bytes/microframe is actually a bit limiting.  So, I was wondering what it would take to make the the library support a high-bandwidth endpoint with the audio device.  

Has anybody tried it?  If not, what is the effort level required to support that?

Thanks!

**Comment by hathach** [comment]
It depends on MCU, though I think high-bandwidth ISO may take quite a lot of effort. You will need to dive into the work to actually know the amount of work

**Reply by ccrome** [reply]
Thanks for the reply Ha!  I was afraid of that.  I guess we can scrape by with 24 or even 16-bit audio if it comes to that.  We can look into it more if/when we need 300 microphones at once :-)

Thanks for the awesome library!

**Comment by ThePonFarr** [comment]
Hi, I am designing a high-bandwidth HID interface with the ESP32-S3 and I was wondering if 1024 byte per packet bulk endpoint is supported? Is it simply a matter of changing this value from 64 to 1024?:

> //--------------------------------------------------------------------
// ENDPOINT FIFO SIZE CONFIGURATION
//--------------------------------------------------------------------
#ifndef CFG_TUD_ENDPOINT0_SIZE
#define CFG_TUD_ENDPOINT0_SIZE 64
#endif
